### ISSUE TITLE
Clarify meaning of clockReference attribute.

### ISSUE THREAD

**Issue by clagms** [modelica/fmi-standard#929]
After discussions with @IZacharias and @KarlWernersson, it became clear that the meaning of the clockReference attribute could be improve.

For now, I only focus on Synchronous Clocks (we haven't yet gone into the details of Communication Point Clocks).

Let C1 denote an input clock, and C2 denote an output clock.
Furthermore, assume that C2 has a clockReference pointing to C1.
AFAIK, the standard allows this since any clock is a variable.

Following the mathematical definition of a clock partition (section 3.1.8.1.1. Synchronous Clocks),
the value of $x_{j-1}$ of the clocked partition associated with clock C1 then consists of a singleton vector with the value of C2 (I presume that the possible values for C2 are "active" and "non-active"). 
I paste the definition here, to help:
![image](https://user-images.githubusercontent.com/871103/79965985-0c4f4c00-848d-11ea-9292-99c4f5917c51.png)

In the same definition, when clock C1 ticks, presumably the value of C2 might change to a new value, which will persist until the next tick of C1.
This is not the same as ticking clock C2 when clock C1 ticks.
Only the interpretation "tick clock C2 when clock C1 ticks" makes sense to us.

However, the ticking of a clock is something that happens in the super dense time domain, and there is no representation for that in the definition of clocked partition.

So, long story short, the meaning of the clockReference attribute is clear when the xml element containing it is a non clock variable, but obscure when the it is a clock variable.

BTW, there is a notion of union in Communication Point Clocks, but that does not help clarify this.

**Comment by IZacharias** [comment]
I think the confusion comes from using the clockReference attribute for different things. For clocks, it is used to define a union of clocks:

> 3.1.8.3.3. Clock Relationships for Communication Point Clocks
> If output clocks and input clocks are defined, it is possible to define a tick relationship from an output clock to an input clock. This is needed to allow for an external resource management to achieve an optimal control of model partitions. The output clocks and the associated input clocks define a union of clocks. In that sense, if an output clock ticks the associated input clock ticks at the same time instant in such a union. It is only meaningful to combine one or multiple output clocks with a single aperiodic input clock in such a union. It is possible to define multiple unions of clocks, with the maximum number defined by the number of available aperiodic input clocks.
{...}
A clock union can be defined based on the clockReference attribute of input clock variables.

To avoid this inconsistant usage of clockReference, I propose to introduce a new attribute „unionClockReference“ for input clock variables.
If an an input clock has a unionClockReference to an output clock, this means that if the output clock ticks the associated input clock ticks at the same time (as described in 3.1.8.3.3).

Furthermore, the clockReference of an output clock should have a similar meaning as it has for all other variables: 
If an output clock has a clockReference to an input clock, this means that the output clock tick event can occur only in the model partition that is related to the input clock, i.e. the output clock’s value (active/inactive) can only change in this model partition.

**Comment by clagms** [comment]
@IZacharias sorry for not having made clear why I didn't mention the clock union. 
I did not mention it because there is no such notion in the Synchronous Clocks (which are the only clocks in Scheduled Co-simulation). 

So the solution you propose might be valid, but we first need to know where the document stands on cascade ticking of Synchronous Clocks (then we can figure out whether the Communication Point clocks are well defined).
In scheduled co-simulation, there can only be synchronous clocks.

**Comment by andreas-junghanns** [comment]
See also #916.
